Cython/Utility/Builtins.c

```diff
@@ -355,6 +355,26 @@ static CYTHON_INLINE int __Pyx_int_from_UCS4(Py_UCS4 uchar);
 //////////////////// int_pyucs4 ////////////////////
 
 static int __Pyx_int_from_UCS4(Py_UCS4 uchar) {
+#if CYTHON_COMPILING_IN_LIMITED_API
+    // Fast path for ascii digits
+    if (likely(uchar >= (Py_UCS4)'0' && uchar <= (Py_UCS4)'9')) {
+        return uchar - (Py_UCS4)'0';
+    }
```

Comment:
Should we always do this? It's the most likely case, probably even worth inlining. CPython goes through a function call otherwise.

Reply:
Yes - now always done. (I haven't inlined it though, mostly out of laziness)